blueprint/proposal/openAPISchemaFromAPICalls.md

```diff
@@ -0,0 +1,30 @@
+
+---
+
+## Overview
+
+The idea proposed in the issue is to be able to generate test cases from API calls. This is certainly a good idea, but if it would have been possible then it would have been done already and used extensively. The reason  openAPI specification exists because the schema is the source of truth for the API.
+
+Also, the task of writing schema is done in the planning phase of the project. It is not something that is done after the project is completed. So, the openAPI spec should already be available before the project is completed. I don't think following a non-conventional reverse approach might be a good idea.
+
+## Problems
+
+- An API call may not define all the possibe fields and resources that are compulsory/required for openAPI schema. If the API call does not define all the fields, then the test case might not be valid, if these are again used to make API calls.
+- `anyOf` and `oneOf` fields can not be generated from API calls. There could be various scenarios where the API call might return a response with one of the possible data types. It would be difficult for the system to detect this.
+- Change in data type may not be detected in the limted API calls made. It could also result if edge cases are not covered.
+- It is difficult to identify the authentication and authorization methods used by the API. openAPI schema has various fields like `BasicAuth`, `BearerAuth`, `ApiKeyAuth`, `OpenID` and `openIdConnect`. It is difficult to identify which of these methods are used by the API.
+- At an abstract level the openAPI schema is a contract between the client and the server. The client is expected to send the request in a certain format and the server is expected to send the response in a certain format. If the API call does not define the format of the request and response, then the openAPI schema might not be valid.
+- An API tester who might have aimed to generate schema may miss on varios edge cases, various data type combinations, etc. They might not be aware of all the possible scenarios and will be limted to the API calls they have made and experience they have had.
+- The openAPI schema has a `maximum` and `minimum` field which can be used to define the upper and lower limits of a field value. This is not possible to be defined in an mock API call.
+- No defination upper and lower limits for fields. This is not possible to be defined in the API call.  For example, a rating system can have a rating between 1 and 5 but this can not be defined in the API call.
+
+These are some of the problems that I can think of. There might be more problems that I am not aware of.
+
+## Conclusion
+
+I think the openAPI schema should be the source of truth for the API. The openAPI schema should be generated in the proper stage of software development life cycle. The openAPI schema should be generated by the API developer and not by the API tester. The API tester should use the openAPI schema to generate test cases.
```

Comment:
Like I have commented earlier it is not about who does it tester/developer majorly we need a less human interactive way to produce openAPI schema.

Reply:
@JammUtkarsh The main motivation behind this project is to make using Keploy a hands free experience. If the user already has an open api schema we will use it, and if they don't, we want to be able to generate the schema on our own so that they don't have to do it. If the developer does not like this behaviour,  they are free to generate their own schema using the tools you mentioned, and then give it to Keploy. But we want the choice to remain with the user.
As far as the problems are concerned, we will have to try and solve them as we move along in the project.